iceoryx_utils/include/iceoryx_utils/internal/units/duration.hpp

```diff
@@ -37,200 +38,311 @@ class Duration;
 
-    /// @brief returns the duration in hours
     template <typename T>
-    constexpr T hours() const;
+    constexpr Duration fromFloatingPointSeconds(const T floatingPointSeconds) const noexcept;
+    template <typename From, typename To>
+    constexpr bool wouldCastFromFloatingPointProbablyOverflow(const From floatingPoint) const noexcept;
 
-    /// @brief returns the duration in days
     template <typename T>
-    constexpr T days() const;
+    constexpr Duration multiplyWith(const std::enable_if_t<!std::is_floating_point<T>::value, T>& rhs) const noexcept;
 
-    /// @brief converts duration in a timeval c struct
-    ///     timeval::tv_sec = seconds since the Epoch (01.01.1970)
-    ///     timeval::tv_usec = microseconds
-    constexpr operator struct timeval() const;
-
-    /// @brief converts time in a timespec c struct
-    struct timespec timespec(const TimeSpecReference& reference = TimeSpecReference::None) const;
-
-    //@ brief Make operators accessible, that have to be defined outside the class
-    // template <typename T>
-    // friend constexpr Duration operator*(const T& left, const Duration& right);
-    // template <typename T>
-    // friend constexpr Duration operator/(const T& left, const Duration& right);
-    friend std::ostream& operator<<(std::ostream& stream, const Duration& t);
-    friend constexpr Duration duration_literals::operator"" _ns(long double);
-    friend constexpr Duration duration_literals::operator"" _ns(unsigned long long int); // PRQA S 48
-    friend constexpr Duration duration_literals::operator"" _us(long double);
-    friend constexpr Duration duration_literals::operator"" _us(unsigned long long int); // PRQA S 48
-    friend constexpr Duration duration_literals::operator"" _ms(long double);
-    friend constexpr Duration duration_literals::operator"" _ms(unsigned long long int); // PRQA S 48
-    friend constexpr Duration duration_literals::operator"" _s(long double);
-    friend constexpr Duration duration_literals::operator"" _s(unsigned long long int); // PRQA S 48
-    friend constexpr Duration duration_literals::operator"" _m(long double);
-    friend constexpr Duration duration_literals::operator"" _m(unsigned long long int); // PRQA S 48
-    friend constexpr Duration duration_literals::operator"" _h(long double);
-    friend constexpr Duration duration_literals::operator"" _h(unsigned long long int); // PRQA S 48
-    friend constexpr Duration duration_literals::operator"" _d(long double);
-    friend constexpr Duration duration_literals::operator"" _d(unsigned long long int); // PRQA S 48
+    template <typename T>
+    constexpr Duration multiplyWith(const std::enable_if_t<std::is_floating_point<T>::value, T>& rhs) const noexcept;
 
   private:
-    /// @brief constructor needs to be private to ensure a unit safe usage of duration
-    constexpr explicit Duration(const long double durationInSeconds);
-    long double durationInSeconds{0.0};
+    SECONDS_TYPE m_seconds{0U};
```

Comment:
I think we should use UpperCamelCase with a postfix _t for type aliases. UPPER_SNAKE_CASE are for compile time constants and enum values.

@elBoberido  And according to my faint and wrinkled memory you introduced also the string type aliases like `ProcessName_t` ... How could you forget this?! ;) 

I need to verify that it is really you, what do you say to this
```cpp
if ( bla )
   while ( fuu )
       doStuff();
```

Please answer with at least two Bobs and a Pöhnl!